src/cmd/cli/command/commands.go

```diff
@@ -53,13 +53,25 @@ func prettyError(err error) error {
 
 }
 
-func detectComposeDownEndLogEventFunc(service string, host string, eventLog string) bool {
-	result := false
-	if service == "cd" && host == "pulumi" {
-		result = strings.Contains(eventLog, "Destroy succeeded in ") ||
-			strings.Contains(eventLog, "Update succeeded in ")
+type EndLogConditional struct {
+	Service  string
+	Host     string
+	EventLog string
+}
+
```

Comment:
Instead of two separate functions to detect compose down and compose up completion this function will create a function which does the checking. The closure will be initialized with the specific log event attributes to look for,